src/app/users/create-user.ts

```diff
@@ -10,21 +9,45 @@ import {
 } from "@services/mongoose"
 import { TwilioClient } from "@services/twilio"
 
-const setupAccount = async (
-  userId: UserId,
-  phoneNumberValid?: PhoneNumber,
-): Promise<Account | ApplicationError> => {
+const setupAccount = async ({
+  userId,
+  config,
+  phoneNumberValid,
+}: {
+  userId: UserId
+  config: AccountsConfig
+  phoneNumberValid?: PhoneNumber
+}): Promise<Account | ApplicationError> => {
   const account = await AccountsRepository().findByUserId(userId)
   if (account instanceof Error) return account
 
-  const wallet = await WalletsRepository().persistNew({
-    accountId: account.id,
-    type: WalletType.Checking,
-    currency: WalletCurrency.Btc,
-  })
-  if (wallet instanceof Error) return wallet
+  const newWallet = (currency: WalletCurrency) =>
+    WalletsRepository().persistNew({
+      accountId: account.id,
+      type: WalletType.Checking,
+      currency,
+    })
+
+  const walletsEnabledConfig = config.wallets.enabledCurrencies
+  const defaultWalletConfig = config.wallets.defaultCurrency
+
+  // Create all wallets
+  const enabledWallets: Partial<Record<WalletCurrency, Wallet>> = {}
+  for (const currency of walletsEnabledConfig) {
+    const wallet = await newWallet(currency)
+    if (wallet instanceof Error) return wallet
+    enabledWallets[currency] = wallet
+  }
 
-  account.defaultWalletId = wallet.id
+  // Set default wallet explicitly, or implicitly as 1st element in
+  // walletsEnabledConfig array.
+  const defaultWalletId =
+    enabledWallets[defaultWalletConfig]?.id || enabledWallets[walletsEnabledConfig[0]]?.id
+
+  if (defaultWalletId === undefined) {
```

Comment:
I think that is an acceptable use case to not have a defaultWalletId ?

like for API/business users (when we eventually get there), you create your account, but you don't necessarily have to have a default wallet at the start.

that said, maybe it's possible our backend doesn't handle yet this use case (?). In that case I would recommand still making it optional to not have a default wallet from the config view point, and we can dive more into the backend if there are some edges cases to fix when the need arise